x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptorTests.java

```diff
@@ -558,10 +557,10 @@ public boolean decRef() {
     public void testSendWithRemoteAccessHeaders() throws Exception {
         assumeTrue("untrusted remote cluster feature flag must be enabled", TcpTransport.isUntrustedRemoteClusterEnabled());
 
-        final Authentication authentication = AuthenticationTestHelper.builder()
-            .user(new User(randomAlphaOfLengthBetween(3, 10), randomRoles()))
-            .realm()
-            .build();
+        final boolean internalUser = randomBoolean();
+        final Authentication authentication = internalUser
+            ? AuthenticationTestHelper.builder().internal(SystemUser.INSTANCE).build()
+            : AuthenticationTestHelper.builder().user(new User(randomAlphaOfLengthBetween(3, 10), randomRoles())).realm().build();
```

Comment:
Do we have a test for throwing exceptions on other internal users?

Reply:
No test, but we have an assertion in the interceptor that covers other internal users. To test this I'd have to toggle off the assertion which feels off. I'll leave as is for now, but I can instead get rid of the assertion and add a test -- let me know if you prefer this (there is still time since this PR is blocked anyways).